packages/rulesets/src/spectral/functions/get-collection-only-has-value-nextlink.ts

```diff
@@ -1,17 +1,27 @@
 // Verifies that collection get schemas have only the `value` and `nextLink` properties in their models.
 
-export const getCollectionOnlyHasValueAndNextLink = (properties: any, _opts: any, ctx: any) => {
-  if (!properties || typeof properties !== "object") {
-    return []
-  }
-  const keys = Object.keys(properties)
-
-  if (keys.length != 2 || !keys.includes("value") || !keys.includes("nextLink")) {
-    return [
-      {
-        message: "Get endpoints for collections of resources must only have the `value` and `nextLink` properties in their model.",
-      },
-    ]
+export const getCollectionOnlyHasValueAndNextLink = (prop: any, _opts: any, ctx: any) => {
+  for (let path of ctx.path) {
+    if (path.includes(".")) {
+      // Splitting the path by Provider namespace
+      let splitNamespace = path.split(".")
+      // finiding number of segments after namespace
+      if (path.includes("/")) {
+        let segments = splitNamespace[1].split("/")
+        if (segments.length % 2 !== 0) {
```

Comment:
>  if (segments.length % 2 !== 0) { [](http://example.com/codeflow?start=6&length=34)

should this not be segments.length % 2 === 0 ?

Reply:
In the first "if" I divided with dot, so half of the namespaces will be one word which is Network from Microsoft.Network
 After "/" more even values i.e. 4 -> dnsZones/{zoneName}/dnssecConfigs/anystring
Microsoft.Network/dnsZones/{zoneName}/dnssecConfigs/anystring - total count will be odd